web/alloy/lights.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+require_once dirname(__FILE__) . '/../../conf/alloy';
+require_once dirname(__FILE__) . '/../fns.php';
+
+$bbox = get_bbox();
+$url = get('url', '.*');
+$token = API_KEYS[$url];
+
+$page = 1;
+while (true) {
+    $results = process_page($token, $bbox, $page);
+    $features = array_merge($features, $results);
+    if (count($results) == 100) {
+        $page++;
+    } else {
+        break;
+    }
+}
+
+print json_encode($geojson);
+
+# ---
+
+function process_page($token, $bbox, $page) {
+    $design = "designs_streetLights";
+    $attributes = ["attributes_itemsTitle", "attributes_itemsGeometry"];
+    $join_attributes = ["root.attributes_streetLightingUnitsUnitType.attributes_streetLightingUnitTypesDescription"];
+    $private_types = [
```

Comment:
I see this list came from the code previously used (and maybe I should have mentioned when reviewing back then!), but might be a good idea to comment where this list came from - is it client-specific, how do we keep it up to date etc.

Reply:
Yeah, the script should probably have Oxfordshire in its name, it's Oxfordshire specific to get the unit type and private nature, couldn't think of an easy way to make it generic at this point.

Reply:
I renamed it, and also factored out the process page to fns to make it more generic, passing in the join vars and the extra data function, let me know how that looks.

Reply:
Looks good 👍